### ISSUE TITLE
[Cygwin 3.3.3] Copy/Paste in SSH is no longer reliable

### ISSUE THREAD

**Issue by hummeleBop** [git-for-windows/git#3634]
Hi there,

When I try to copy some text from the host session and past to the ssh (provided by git for windows) remote session, parts are missing.

It works fine in Git for windows 2.34.1

Thanks

**Comment by dscho** [comment]
> I try to copy some text from the host session and past to the ssh (provided by git for windows) remote session, parts are missing.

Works here.

**Comment by hummeleBop** [comment]
First I push a string in my clipboard, like that : 

```
cat - <<-'ABC'
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed malesuada elit ut porttitor fringilla. Duis vestibulum hendrerit nibh, vitae posuere orci malesuada non. Donec eu risus tellus. Aenean lacinia lacinia semper. Sed fermentum mi ut turpis porttitor fermentum. Aenean sapien nulla, accumsan vitae ligula non, venenatis egestas nulla. Pellentesque malesuada nisl non mi laoreet suscipit. Aliquam eu blandit erat. Mauris volutpat vehicula neque, nec vestibulum sapien pharetra sit amet. Curabitur in fermentum lacus. Praesent hendrerit massa sit amet urna aliquam, quis bibendum ex mattis. Proin porttitor condimentum tellus, quis condimentum nibh varius a. Aliquam rutrum ex nec hendrerit ultrices. Donec justo justo, pellentesque eget imperdiet et, lobortis scelerisque dolor. Aliquam pulvinar mauris eget ipsum eleifend, vel blandit elit feugiat. Suspendisse aliquam nisi nec nisl finibus, eu viverra mauris lacinia.

Nullam tincidunt risus sem, ut pharetra ligula aliquam et. Integer tincidunt magna mi, eu ultrices nibh facilisis a. Praesent vestibulum sit amet elit in semper. In hac habitasse platea dictumst. Etiam sagittis sit amet nisi id laoreet. Etiam aliquam convallis tincidunt. Nunc nunc ipsum, venenatis vel turpis ut, convallis sodales enim. In neque metus, sodales sit amet nunc sed, ultrices consequat quam. Donec ac augue aliquet, luctus massa a, fringilla erat. Aliquam at magna vitae sem aliquet tincidunt. Sed tempus, eros sed accumsan accumsan, nisi neque euismod diam, sed mattis felis quam eget metus. Fusce id neque id turpis blandit rhoncus. In hac habitasse platea dictumst. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas.

Mauris vitae nibh tristique, venenatis ligula in, egestas magna. Fusce pulvinar feugiat ante vitae venenatis. Etiam pretium rutrum justo, quis pretium urna luctus ac. Cras lacus risus, varius et pretium lobortis, feugiat sit amet neque. Phasellus et velit metus. Sed ultricies sodales sollicitudin. Nunc nisi turpis, condimentum vitae rutrum eget, pretium ac libero. Vivamus cursus semper metus elementum consectetur.

Nunc ullamcorper luctus tortor id posuere. Sed condimentum condimentum dolor eget laoreet. Aenean lacinia enim nec tellus varius pellentesque. Sed eu justo interdum, maximus turpis at, fermentum lectus. Nullam ac bibendum odio. Aenean felis justo, consectetur vel rhoncus faucibus, ultricies vitae lectus. Mauris nec est eros. Ut accumsan tincidunt libero, eget aliquam nisl maximus in. Vivamus enim erat, accumsan vel varius sit amet, vestibulum quis purus. Phasellus tincidunt tincidunt risus eget maximus. Vestibulum pretium sapien sit amet tortor euismod vulputate. Donec suscipit mauris vitae ligula facilisis porttitor.

Nunc commodo vestibulum convallis. Sed eleifend arcu sed turpis porttitor feugiat. Donec molestie, libero ut sollicitudin congue, nisl felis rutrum enim, ac blandit ligula diam vitae nunc. Curabitur diam dolor, ornare sed velit non, vehicula imperdiet magna. Morbi nunc magna, luctus ut rutrum sed, feugiat nec nisl. Nam porta, arcu sed ultricies placerat, metus metus vulputate tellus, non vestibulum dolor felis nec ante. Fusce in finibus lacus. Nullam varius euismod leo, sed consequat nunc cursus nec. Proin tincidunt, turpis eu egestas sollicitudin, lacus quam ultrices lorem, in consequat tellus tortor ut leo. Maecenas at risus ut ipsum condimentum egestas. Donec fringilla finibus ipsum et mollis. Sed accumsan urna convallis condimentum fermentum. Phasellus feugiat dictum leo, in pellentesque nulla ultrices et. Fusce sit amet finibus mauris, nec tristique nisl. Mauris iaculis eu lorem quis aliquam.
ABC
```

Then I paste that in my SSH session. It works fine sometimes, but I also get this result : 


```
$. Integer tincidunt magna mi, eu ultrices nibh facilisis a. Praesent vestibulum sit amet elit in semper. In hac habitasse platea dictumst. Etiam sagittis sit amet nisi id laoreet. Etiam aliquam convallis tincidunt. Nunc nunc ipsum, venenatis vel turpis ut, convallis sodales enim. In neque metus, sodales sit amet nunc sed, ultrices consequat quam. Donec ac augue aliquet, luctus massa a, fringilla erat. Aliquam at magna vitae sem aliquet tincidunt. Sed tempus, eros sed accumsan accumsan, nisi neque euismod diam, sed mattis felis quam eget metus. Fusce id neque id turpis blandit rhoncus. In hac habitasse platea dictumst. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas.
-bash: Integer: No such file or directory
$
$Mauris vitae nibh tristique, venenatis ligula in, egestas magna. Fusce pulvinar feugiat ante vitae venenatis. Etiam pretium rutrum justo, quis pretium urna luctus ac. Cras lacus risus, varius et pretium lobortis, feugiat sit amet neque. Phasellus et velit metus. Sed ultricies sodales sollicitudin. Nunc nisi turpis, condimentum vitae rutrum eget, pretium ac libero. Vivamus cursus semper metus elementum consectetur.
-bash: Mauris: command not found
$
$Nunc ullamcorper luctus tortor id posuere. Sed condimentum condimentum dolor eget laoreet. Aenean lacinia enim nec tellus varius pellentesque. Sed eu justo interdum, maximus turpis at, fermentum lectus. Nullam ac bibendum odio. Aenean felis justo, consectetur vel rhoncus faucibus, ultricies vitae lectus. Mauris nec est eros. Ut accumsan tincidunt libero, eget aliquam nisl maximus in. Vivamus enim erat, accumsan vel varius sit amet, vestibulum quis purus. Phasellus tincidunt tincidunt risus eget maximus. Vestibulum pretium sapien sit amet tortor euismod vulputate. Donec suscipit mauris vitae ligula facilisis porttitor.
-bash: Nunc: command not found
$
$Nunc commodo vestibulum convallis. Sed eleifend arcu sed turpis porttitor feugiat. Donec molestie, libero ut sollicitudin congue, nisl felis rutrum enim, ac blandit ligula diam vitae nunc. Curabitur diam dolor, ornare sed velit non, vehicula imperdiet magna. Morbi nunc magna, luctus ut rutrum sed, feugiat nec nisl. Nam porta, arcu sed ultricies placerat, metus metus vulputate tellus, non vestibulum dolor felis nec ante. Fusce in finibus lacus. Nullam varius euismod leo, sed consequat nunc cursus nec. Proin tincidunt, turpis eu egestas sollicitudin, lacus quam ultrices lorem, in consequat tellus tortor ut leo. Maecenas at risus ut ipsum condimentum egestas. Donec fringilla finibus ipsum et mollis. Sed accumsan urna convallis condimentum fermentum. Phasellus feugiat dictum leo, in pellentesque nulla ultrices et. Fusce sit amet finibus mauris, nec tristique nisl. Mauris iaculis eu lorem quis aliquam.
-bash: Nunc: command not found
$ABC
-bash: ABC: command not found
$~

```
It looks like the buffer is cut before it is totally sent to the remote session.

**Comment by dscho** [comment]
Thank you for the more verbose reproducer. However, it still does not reproduce for me. Could you try this with [-rc1](https://github.com/git-for-windows/git/releases/tag/v2.35.0-rc1.windows.1)? I incorporated an MSYS2 runtime fix that I thought was unrelated, but could potentially have fixed your issue (your report unfortunately cut the template rather than filling it out, therefore I cannot tell).

**Comment by hummeleBop** [comment]
The RC1 has fixed this peculiar issue but the emojis no longer work now..

![image](https://user-images.githubusercontent.com/3150838/149946119-07a6bcd2-eefe-482d-935d-5b647ba09f08.png)

**Comment by dscho** [comment]
Seems to work quite well here:

![image](https://user-images.githubusercontent.com/127790/150209611-ad880943-8f3a-4bcf-b7e6-4f1f92dda159.png)

If you had invested some love in your report to include vital information such as the terminal you are using, that might have made things a bit clearer. As it stands, I spent all the time I am willing on this ticket.

**Comment by hummeleBop** [comment]
Yea sorry, I was using Windows terminal. But you can get rid of this ticket.

**Comment by dscho** [comment]
Seems that the _new_ issue you wanted to report is the same as was reported in https://github.com/git-for-windows/git/issues/3281, fixed in https://github.com/git-for-windows/git/issues/3541, and regressed since.

**Comment by dscho** [comment]
> Seems that the _new_ issue you wanted to report is the same as was reported in #3281, fixed in #3541, and regressed since.

And a work-around is to enable pseudo console support (it's in the installer, in the Experimental Options, on the last options page before the installation starts).

**Comment by dscho** [comment]
Closing as stale.
